src/main/java/core/mate/academy/service/MachineService.java

```diff
@@ -17,22 +17,23 @@
      * @param type - any class of Machine sub class. For example: Track.class or Bulldozer.class
      * @return the list of machines
      */
-    List<Machine> getAll(Class type);
+    List<Machine> getAll(Class<? extends Machine> type);
 
     /**
      * Fill the machines list with passed value
      * Replace the Object with parametrized value.
      * This method should be able to work well with any type of machine passed as 'value'
+     *
      * @param machines - list of machines to be filled with value
-     * @param value - any object of machine sub class
+     * @param value    - any object of machine sub class
      */
-    void fill(List<Object> machines, Object value);
+    void fill(List<? super T> machines, T value);//? super Machine
```

Comment:
```suggestion
    void fill(List<? super T> machines, T value);
```
you wrote correct, let's use T here, so we will stick to the Type assigned during class declaration